src/utils/withDeprecationWarning/index.js

```diff
@@ -1,8 +1,15 @@
+const allowedEnvironments = ['test', 'development', 'dev'];
+const conditional1 = allowedEnvironments.includes(process.env.NODE_ENV);
```

Comment:
could we rename this variable to something more descriptive like `shouldShowForEnvironment`